tests/idris2/evaluator005/Issue1200.idr

```diff
@@ -0,0 +1,9 @@
+module Issue1200
+
+%default total
+
+main : IO ()
+main = traverse_ printLn $ do
+  f <- [(1/), log, sin, cos, tan, asin, acos, atan, sqrt, floor, ceiling]
+  x <- [-2,-1,-0.5,0,0.5,1,2]
+  pure (f x)
```

Comment:
Can change to `traverse_ printLn $ compare expected $ do...` where:
```idris
data Result = UpTo Double Double | PosInf | NaN ...
||| check whether the double is in the list of results
check : List Result -> Double -> Bool

expected : List [Result] -> List Double -> List Bool
expected results = map check . zip results
```

Reply:
The point made in #623 is that different results in different backends will lead
to issues (runtime crashes in the best cases) so I don't think we want to write a
test case that is explicitly accepting of such a broken state of affairs.

I wonder whether this could be a version problem. Brew tends to install newer versions
AFAICT. I'll check after our weekly meeting.

Reply:
I see. #623 says the result is non-deterministic, based on optimisation levels and architecture, so as @wenkokke said, we're going to get different behaviour between type-checking time and runtime, which you can bootstrap to runtime inconsistencies.

The #601 shows that the incongruence of floats means they're unsafe for type-checking time anyway.

Maybe all floating point operations should be monadic?

Reply:
It is worse than merely compile-time versus runtime. The same compiled program can arrive at different results for the same floating point computation, on the same computer, run at different times. That's because the settings of the FPU can be changed at runtime! For example, control-F for "6.2.3.3  Rounding Control" in the [AMD x86_64 architecture manual](https://www.amd.com/system/files/TechDocs/24592.pdf). That's right - the x87 rounding mode is set by a couple of register bits, and can be changed any time!

This leads us to a serious dilemma. Since floating point operations depend on a part of CPU state, there is no such thing as a pure floating point function. I guess that means it should be monadic like @ohad said (I am a beginner with functional programming concepts so I can't say much about it). Nonetheless, I _deeply_ and _sincerely_ hope that we can keep writing code that processes Doubles in a natural way.